test/functional/services/dashboard/add_panel.js

```diff
@@ -172,7 +177,12 @@ export function DashboardAddPanelProvider({ getService, getPageObjects }) {
     }
 
     async filterEmbeddableNames(name) {
-      await testSubjects.setValue('savedObjectFinderSearchInput', name);
+      // The search input field may be disabled while the table is loading so wait for it
+      await this.waitForEuiTableLoading();
+      // This try loop lets us recover it the field wasn't enabled yet
+      await retry.try(async () => {
+        await testSubjects.setValue('savedObjectFinderSearchInput', name);
```

Comment:
I don't think this extra retry loop adds anything because there is already a retry loop on the inside of setValue:

```
async setValue(selector, text) {
      return await retry.try(async () => {
        await this.click(selector);
        // in case the input element is actually a child of the testSubject, we
        // call clearValue() and type() on the element that is focused after
        // clicking on the testSubject
        const input = await remote.getActiveElement();
        await input.clearValue();
        await input.type(text);
      });
    }
```

We could think about confirming that the value was actually set by adding something in that function call that does like:

```
async setValue(selector, text) {
      return await retry.try(async () => {
        await this.click(selector);
        // in case the input element is actually a child of the testSubject, we
        // call clearValue() and type() on the element that is focused after
        // clicking on the testSubject
        const input = await remote.getActiveElement();
        await input.clearValue();
        await input.type(text);
        const newValue = await this.getVisibleText(selector);
         if (newValue != text) {
           throw new Error('Value was not set');
          } 
      });
    }
```

But I'm not sure if `getVisibleText` is what we'd want or if it's work on all element types (might be something like getValue we should use?).

Reply:
I didn't realize that setValue method had a retry loop.  But that makes me wonder why it wasn't solving this problem from the beginning.  The first error in this case was coming from the clearValue() step in that setValue method;
`invalid element state: Element must be user-editable in order to clear it.`

But then the next message says that it failed to find the element.
`no such element: Unable to locate element: {"method":"css selector","selector":"[data-test-subj~="savedObjectFinderSearchInput"]"}`

And when the test finally fails and takes a screenshot, we see that it has navigated away from the Add Panel flyout and has opened the New visualization page.  It *seems* as if this inner-most retry isn't staying in this loop for some reason, and an outer retry is trying to clicking on "Add" again, or somewhere else that causes the incorrect navigation.

What you don't see here is the debug logging I had in place which did show that the euiBasicTable did initially show the `euiBasicTable-loading` class every time we opened the Add Panel flyout and that it was gone after we waited for it to be deleted;

```
         │ debg  DashboardAddPanel.isAddPanelOpen
         │ debg  TestSubjects.exists(dashboardAddPanel)
         │ debg  existsByDisplayedByCssSelector [data-test-subj~="dashboardAddPanel"]
         │ debg  TestSubjects.find(dashboardAddPanel)
         │ debg  findByCssSelector [data-test-subj~="dashboardAddPanel"]
         │ debg  ################################################################# euiBasicTable euiBasicTable-loading
         │ debg  ##############################++++++++++++++++++++++++++++++++++++++ euiBasicTable
         │ debg  TestSubjects.find(savedObjectFinderSearchInput)
         │ debg  findByCssSelector [data-test-subj~="savedObjectFinderSearchInput"]
         │ debg  isGlobalLoadingIndicatorVisible
         │ debg  TestSubjects.exists(globalLoadingIndicator)
```

Reply:
Yea, so my suspicion is that the mechanism for typing selenium uses is accidentally hitting the `Add Visualization` button.  Once the page navigates away, doesn't matter how many times it loops, it's going to fail. That's why I think we have to add a function that waits till the flyout finishes it's transition before we even start typing into that field.